File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/TP6 - Warbot/warbot-3.3.3/teams/javascript_level_0/WarRocketLauncher.js
var RocketLauncher = JsWarRocketLauncher.extend({

	action : function() { 

		if(this.isBlocked()) {
			this.randomHeading();
		}

		
		return this.move(); 

	}
});


var WarRocketLauncher = new RocketLauncher();

File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/TP6 - Warbot/warbot-3.3.3/teams/python_test/WarRocketLauncher.py

def actionWarRocketLauncher():

	# Les deux envois de message suivant ne marchent pas !
	#broadcastAll("broadcastAll","content")
	#broadcastMessageToAll("broadcastMessageToAll",["content1","content2"])
	broadcastMessageToAgentType(WarAgentType.WarRocketLauncher,"broadcastMessageToAgentType",["content3","content4"])
	if(isBlocked()):
		RandomHeading()
	return move();
